server/controllers/user.controller.ts

```diff
@@ -1,8 +1,109 @@
+// import { IUser } from "../interfaces/user.interface";
+
+// const register = async (req: Request, res: Response) => {}
+// const login = async (req: Request, res: Response) => {}
+// const logout = async (req: Request, res: Response) => {}
+// const getOneUser = async (req: Request, res: Response) => {}
+
+// export default { register, login, logout, getOneUser }
+
 import { Request, Response } from "express";
+import User from "../models/user.model"
+const bcrypt = require('bcrypt');
+const jwt = require("jsonwebtoken");
+
+module.exports = {
+  register: (req: Request, res: Response) => {
+    console.log("in register");
+    console.log(req.body);
+    const user = new User(req.body);
+
+    user.save()
+      .then((newUser) => {
+        console.log(newUser);
+        console.log("registration successful");
+        res.json({
+          successMessage: "Thank you for registering!",
+          user: newUser,
+        });
+      })
+      .catch((err: unknown) => {
+        console.log("registration unsuccessful");
```

Comment:
Don't forget to remove the console.logs as per kishan mentioned

Reply:
done, thanks